### DISCUSSION TITLE
How to kill a long vasp task in the dpgen process?

### DISCUSSION THREAD

**Discussion by Guancred** [deepmodeling/dpgen#1427] [Q&A]
Hi,
when i run the single point energy calculations during the dpgen process, there may be a task will take a very long time (may be 4 hours even 10 hours), so how to kill such task?

**Comment by njzjz** [comment]
Perhaps you can use some Linux tools like [`timeout`](https://man7.org/linux/man-pages/man1/timeout.1.html) to do it.

**Reply by Guancred** [reply]
you mean i can set the keywork of "time_limte" in the machine.json? as the figure 
![image](https://github.com/deepmodeling/dpgen/assets/91767936/df40b0f3-81cd-4f0a-a69a-af1ba754d9be)

**Reply by njzjz** [reply]
Setting the command to something like `timeout 1h vasp_std || :` so it will be killed if the program runs for more than 1h.

**Reply by Guancred** [reply]
I set the command like "timeout 20s vasp_std || mpirun -np 40 vasp_std" in the machine.json, it does not work, while when i set "timeout 20s vasp_std && mpirun -np 40 vasp_std", the task was killed successfully. But the dpdispatcher will try to resubmit the task and an error occurred:
![1FAZ7`A{H(Q_BA }(D1${7Q](https://github.com/deepmodeling/dpgen/assets/91767936/bee65b83-916c-4733-8ff0-fae5ff91edd7)

**Reply by njzjz** [reply]
That's what `|| :` is for.
